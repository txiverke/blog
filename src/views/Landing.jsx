// @flow

import React from 'react'
import Helmet from 'react-helmet'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

import Background from '../components/Background'
import Loading from '../components/Loading'
import { getSlug, normalizeVal } from '../utils/helpers'
import { loadStats } from '../actionCreators'

const sections = [
  { name: 'About me' },
  { name: 'Posts' },
  { name: 'Projects' }
]

class Landing extends React.Component {
  props: {
    match: Object,
    location: Object,
    history: Object,
    staticContext: any,
    dispatch: Function,
    statistic: Object,
  }
  componentDidMount() {
    const { dispatch, statistic } = this.props

    if (Object.keys(statistic.data).length === 0) {
      dispatch(loadStats())
    }
  }

  render() {
    const completed = this.props.statistic.completed
    const getValue = (section) => {
      const { users, posts } = this.props.statistic.data

      switch (section) {
        case 'about-me': return normalizeVal(users)
        case 'posts': return normalizeVal(posts)
        default: return '00'
      }
    }

    if (completed) {
      return (
        <section className="app-view">
          <Helmet title="Welcome!" />
          <Background title="Just a tes of bg" name="bg_1.png" {...this.props} />
          <div className="app-content">
          {sections.map(item => {
            const name = getSlug(item.name)
            return (
              <article key={item.name} className="app-content-landing">
                <h2>
                  <Link to={name}>
                    <span>{getValue(name)}</span>
                    <span className="bold">{item.name.toUpperCase()}</span>
                  </Link>
                </h2>
              </article>
            )
          })}
          </div>
        </section>
      )
    }

    return <Loading msg={this.props.statistic.message} />
  }
}

const mapStateToProps = state => ({ statistic: state.statistic })

export default connect(mapStateToProps)(Landing)