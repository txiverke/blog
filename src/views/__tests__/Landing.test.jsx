import React from 'react'
import { Unwrapped as UnwrappedLanding } from '../Landing'
import { shallow } from 'enzyme'

describe('<Landing />', () => {
  it('should render itself', () => {
    const language = { current: 'eng' }
    const component = shallow(<UnwrappedLanding language={language} />)
    expect(component).toMatchSnapshot()
  })
})