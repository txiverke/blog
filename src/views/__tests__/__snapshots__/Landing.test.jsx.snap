// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    language={
        Object {
            "current": "eng",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
          meta={
                    Array [
                              Object {
                                "content": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
                                "name": "description",
                              },
                              Object {
                                "content": "Welcome to  xaviervila.tech",
                                "property": "og:title",
                              },
                            ]
          }
          title="Welcome to  xaviervila.tech"
/>,
        <Background
          title="Landing page"
/>,
        <Connect(Stats)
          DIC={
                    Object {
                              "ABOUT-ME": "ABOUT ME",
                              "BACK": "Back",
                              "BACK_TO": "Back to the landing page",
                              "CONTACT": "CONTACT",
                              "DESCRIPTION": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
                              "DOWNLOADED_RESUME": "Résumé downloaded",
                              "DOWNLOAD_RESUME": "Download Résumé",
                              "EDIT": "EDIT",
                              "FRONTEND": "Front-end developer",
                              "NOT_EXIST": " does not exist.",
                              "NO_POSTS_MSG": "Press a tag button to see content.",
                              "PAGE_NOT_FOUND": "Page not found",
                              "POSTED_AT": "Posted at ",
                              "POSTS": "POSTS",
                              "PROJECTS": "PROJECTS",
                              "RESUME": "Thanks for downloading my Résumé.",
                              "SORRY": "Sorry, but the url: ",
                              "WELCOME": "Welcome to ",
                            }
          }
/>,
      ],
      "className": "app-view app-view-centered",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defer": true,
          "encodeSpecialCharacters": true,
          "meta": Array [
            Object {
              "content": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
              "name": "description",
            },
            Object {
              "content": "Welcome to  xaviervila.tech",
              "property": "og:title",
            },
          ],
          "title": "Welcome to  xaviervila.tech",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Landing page",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "DIC": Object {
            "ABOUT-ME": "ABOUT ME",
            "BACK": "Back",
            "BACK_TO": "Back to the landing page",
            "CONTACT": "CONTACT",
            "DESCRIPTION": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
            "DOWNLOADED_RESUME": "Résumé downloaded",
            "DOWNLOAD_RESUME": "Download Résumé",
            "EDIT": "EDIT",
            "FRONTEND": "Front-end developer",
            "NOT_EXIST": " does not exist.",
            "NO_POSTS_MSG": "Press a tag button to see content.",
            "PAGE_NOT_FOUND": "Page not found",
            "POSTED_AT": "Posted at ",
            "POSTS": "POSTS",
            "PROJECTS": "PROJECTS",
            "RESUME": "Thanks for downloading my Résumé.",
            "SORRY": "Sorry, but the url: ",
            "WELCOME": "Welcome to ",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            meta={
                        Array [
                                    Object {
                                      "content": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
                                      "name": "description",
                                    },
                                    Object {
                                      "content": "Welcome to  xaviervila.tech",
                                      "property": "og:title",
                                    },
                                  ]
            }
            title="Welcome to  xaviervila.tech"
/>,
          <Background
            title="Landing page"
/>,
          <Connect(Stats)
            DIC={
                        Object {
                                    "ABOUT-ME": "ABOUT ME",
                                    "BACK": "Back",
                                    "BACK_TO": "Back to the landing page",
                                    "CONTACT": "CONTACT",
                                    "DESCRIPTION": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
                                    "DOWNLOADED_RESUME": "Résumé downloaded",
                                    "DOWNLOAD_RESUME": "Download Résumé",
                                    "EDIT": "EDIT",
                                    "FRONTEND": "Front-end developer",
                                    "NOT_EXIST": " does not exist.",
                                    "NO_POSTS_MSG": "Press a tag button to see content.",
                                    "PAGE_NOT_FOUND": "Page not found",
                                    "POSTED_AT": "Posted at ",
                                    "POSTS": "POSTS",
                                    "PROJECTS": "PROJECTS",
                                    "RESUME": "Thanks for downloading my Résumé.",
                                    "SORRY": "Sorry, but the url: ",
                                    "WELCOME": "Welcome to ",
                                  }
            }
/>,
        ],
        "className": "app-view app-view-centered",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defer": true,
            "encodeSpecialCharacters": true,
            "meta": Array [
              Object {
                "content": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
                "name": "description",
              },
              Object {
                "content": "Welcome to  xaviervila.tech",
                "property": "og:title",
              },
            ],
            "title": "Welcome to  xaviervila.tech",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Landing page",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "DIC": Object {
              "ABOUT-ME": "ABOUT ME",
              "BACK": "Back",
              "BACK_TO": "Back to the landing page",
              "CONTACT": "CONTACT",
              "DESCRIPTION": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
              "DOWNLOADED_RESUME": "Résumé downloaded",
              "DOWNLOAD_RESUME": "Download Résumé",
              "EDIT": "EDIT",
              "FRONTEND": "Front-end developer",
              "NOT_EXIST": " does not exist.",
              "NO_POSTS_MSG": "Press a tag button to see content.",
              "PAGE_NOT_FOUND": "Page not found",
              "POSTED_AT": "Posted at ",
              "POSTS": "POSTS",
              "PROJECTS": "PROJECTS",
              "RESUME": "Thanks for downloading my Résumé.",
              "SORRY": "Sorry, but the url: ",
              "WELCOME": "Welcome to ",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
